src/apps/appstore/app.js

```diff
@@ -299,11 +301,41 @@ define(function(require) {
 							errorCallback && errorCallback();
 						}
 					});
+				},
+
+				appStorePopUpButtonDisable = function(disabled) {
+					parent.find('#appstore_popup_save').prop('disabled', disabled);
+				},
+
+				appStorePopUpDirtyCheck = function() {
+					var allowedUsers = parent.find('.permissions-bloc input[name="permissions"]:checked').val(),
+						selectedUsers = (allowedUsers === 'specific') ? (monster.ui.getFormData('app_popup_user_list_form').users || []) : app.users;
+
+					selectedUsers = _.reduce(selectedUsers, function(users, id) {
```

Comment:
A simple `lodash#map` should achieve the same result, without concatenating arrays for each operations.